proj/src/main/webapp/test/chat.test.js

```diff
@@ -14,6 +14,77 @@ require("../app.js");
 let testHelper = require('./test-helper-functions.js');
 let chat = require('../chat.js');
 
+describe('creating and joining 1-on-1 chats', function() {
+    this.timeout(10000);
+    let testUsers = [{
+        email: 'test@test.com', 
+        password: 'test123!'
+    }, 
+    {
+        email: 'test2@test.com', 
+        password: 'test123!'
+    }];
+    let chatId;
+    const oneOnOneRef = firebase.database().ref('/chat/chats-1on1');
+
+    before(async function() {
+        for (let i = 0; i < testUsers.length; i++) {
+            await testHelper.createUserIfNotExisting(testUsers[i]);
+            await firebase.auth().signInWithEmailAndPassword(testUsers[i].email, testUsers[i].password);
+            testUsers[i]["uid"] = firebase.auth().currentUser.uid;
```

Comment:
Check out `testHelper.createUserIfNotExisting`. It updates the testUser objects to include the uid after creating the user and it leaves the user signed in when it finishes. So you will probably need to sign out afterward so that the next user can properly sign in and have it's uid logged. I guess you have `if (i==1) {...} else { sign out }` below

Reply:
yeah I have to leave one user logged in because the function i'm testing uses firebase.auth().currentUser.uid